packages/logging/src/log-escalator.ts

```diff
@@ -0,0 +1,96 @@
+import { Logger } from 'typescript-log'
+
+export interface EscalatingLog extends Logger {
+    /** Flushes the logs to the underlying logger */
+    done(): void
+}
+
+export function logEscalator(
+    log: Logger,
+    {
+        setTimeout: setTimeoutOption,
+        clearTimeout: clearTimeoutOption,
+        timeout = 10000,
+    }: {
+        setTimeout?: (cb: () => void, timeout: number) => NodeJS.Timeout
+        clearTimeout?: (timeout: NodeJS.Timeout) => void
+        timeout?: number
+    } = {},
+): EscalatingLog {
+    let isDone = false
+    const logBuffer: Array<{ logMethod: string; args: any[] }> = []
+    const timedOut: NodeJS.Timeout = (setTimeoutOption || (setTimeout as any))(done, timeout)
+    timedOut.unref()
+
+    return {
+        child() {
+            throw new Error('Log escalator does not support creating child loggers')
```

Comment:
You could include a hint in the error message to create the escalator on the child logger instead, as per https://github.com/sevenwestmedia-labs/etrigan/pull/22/files#r404468970

Reply:
Done